WordPress/Classes/System/WindowManager.swift

```diff
@@ -0,0 +1,86 @@
+import Foundation
+import WordPressAuthenticator
+
+/// This class takes care of managing the App window and its `rootViewController`.
+/// This is mostly intended to handle the UI transitions between authenticated and unauthenticated user sessions.
+///
+@objc
+class WindowManager: NSObject {
+
+    /// The App's window.
+    ///
+    private let window: UIWindow
+
+    /// A boolean to track whether we're showing the sign in flow.
+    ///
+    private var isSignInVisible = false
+
+    init(window: UIWindow) {
+        self.window = window
+    }
+
+    // MARK: - Initial App UI
+
+    /// Shows the initial UI for the App to be shown right after launch.  This method will present the sign-in flow if the user is not
+    /// authenticated, or the actuall App UI if the user is already authenticated.
+    ///
+    func showUI() {
+        if AccountHelper.isLoggedIn {
+            showUIForAuthenticatedUsers()
+        } else {
+            showUIForUnauthenticatedUsers()
+        }
+    }
+
+    /// Shows the SignIn UI flow if the conditions to do so are met.
+    ///
+    @objc
+    func showSignInIfNecessary() {
+        guard isSignInVisible == false && AccountHelper.isLoggedIn == false else {
+            return
+        }
+
+        showUIForUnauthenticatedUsers()
+    }
+
+    /// Shows the UI for authenticated users.
+    ///
+    func showUIForAuthenticatedUsers() {
+        isSignInVisible = false
+
+        show(WPTabBarController.sharedInstance())
+    }
+
+    /// Shows the initial UI for unauthenticated users.
+    ///
+    func showUIForUnauthenticatedUsers() {
+        isSignInVisible = true
+
+        guard let loginViewController = WordPressAuthenticator.loginUI() else {
```

Comment:
we might also want to track `.openedLogin` like in the previous `WordPressAuthenticator.showLogin`?

Reply:
Thanks!

Reply:
I've added the tracking call.

I was thinking though that maybe this tracking call should be in the VC's `viewWillAppear` method in the Authenticator pod?  I think it could be a less  "fragile" approach in sight of future changes.

Definitely not for the scope of this PR though, but let me know what you think.

Reply:
Moving the tracking call to the login VC's `viewWillAppear` sounds like a good idea! I can make this change after code freeze for the authenticator library later this week 👍 just created an issue: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/issues/567

Reply:
Thanks!